### ISSUE TITLE
Consultation on snATAC-seq Data Analysis for differential peak analysis

### ISSUE THREAD

**Issue by RuiyuRayWang** [beyondpie/CEMBA_wmb_snATAC#12]
**TL;DR**
In doing differential peak analysis,
- what's the difference between `union_bed_file` and `rnd_bed_file`
- which to use for `pp.make_peak_matrix`

**Original post**

Hi, @beyondpie. Thanks for making this remarkable pipeline publicly available.

I'd like to work with processed files from [catlas.org](http://catlas.org/wholemousebrain/), subset out my cell types of interest, create a peak matrix and perform differential peak analysis.

I downloaded files with the `_rm_dlt.h5ad` suffix under [`snapatac2_anndata_per_sample_with_fragment_dir`](http://catlas.org/renlab_downloads/wholemousebrain/snapatac2_anndata_per_sample_with_fragment_dir/), and used the following code to get peak matrix:

> peak_mat = sa2.pp.make_peak_matrix(
>        adata = snap2,
>        inplace = False,
>        file = outdir / "sa2pmat.union_pmat.h5ad",
>        backend = "hdf5",
>        peak_file = union_bed_file,
>        chunk_size = 10000,
>        use_x = False
>        )
(ref. [sa2pmay.py lines 66-72](https://github.com/beyondpie/CEMBA_wmb_snATAC/blob/main/03.peakcalling/src/main/python/sa2pmat.py#L66))

where
> union_bed_file = 'mba.whole.sa2.final.peak.srt.bed'

Then do differential peak analysis with [`tl.marker_regions` or `tl.diff_test`](https://kzhang.org/SnapATAC2/version/2.4/tutorials/diff.html#Finding-marker-regions).

however I'm unsure if this is correct as I noticed two peak files:
`union_bed_file` and `rnd_bed_file`, where
> rnd_bed_file = 'mba.whole.shuffle.removeovlp.bed'

What's the purpose of having both bed files and which one should I use for diff-peak analysis?

Thanks!
Ray

**Comment by beyondpie** [comment]
Hi Ray,

1. `rnd_bed_file` is used for our analysis, such as Cicero, as a random control. So you can ignore it. 
2. `union_bed_file` should have all the peaks we identified (merged from all the cell types).
3. I also shared the pmat, gmat, bmat information here:https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE246791
4. `pp.make_peak_matrix` is used to create pmat, i.e, single-cell level counts on peak regions.


Thanks!
Songpeng

**Comment by RuiyuRayWang** [comment]
Thank you Songpeng! I'll work with `union_bed_file` and see how it goes.

**Comment by RuiyuRayWang** [comment]
Hi @beyondpie ,

Following your advice, I used `union_bed_file` to obtain pmat and successfully performed differential peak analysis. But here's a follow-up question:

The peaks in the `union_bed_file` were given by a fixed width of 500bp. Consequently, peaks in the differential analysis were also 500bp in width. How do I obtain accurate peak (cCRE) boundaries?
I'm considering plotting reads coverage grouped by cell types, so that I can visualize the peaks more intuitively, but snapatac2 currently doesn't provide that utility. Do you know how this might be done?

Thanks!

Best regards,
Ray

**Comment by RuiyuRayWang** [comment]
> Do you mean you want to use IGV to visualize the tracks (as bigwig format? If so, you can try this function: https://kzhang.org/SnapATAC2/api/_autosummary/snapatac2.ex.export_coverage.html)

> BTW, in CATLas, we have uploaded bigwig tracks on subclass level:
http://catlas.org/wholemousebrain/#!/igv, hope this works on your side. (Feel free to download the bigwig files we generated on subclass level.)

Thank you Songpeng! I think I'll try out these suggestions and see how it goes.

Best,
Ray

**Comment by beyondpie** [comment]
@RuiyuRayWang 
Should I close this issue?
Songpeng

**Comment by RuiyuRayWang** [comment]
> @RuiyuRayWang Should I close this issue? Songpeng

Yes please
